ci/run_clang_tidy.sh

```diff
@@ -26,7 +26,7 @@ echo "Generating compilation database..."
 
 # bazel build need to be run to setup virtual includes, generating files which are consumed
 # by clang-tidy
-"${ENVOY_SRCDIR}/tools/gen_compilation_database.py" --include_headers
+"${ENVOY_SRCDIR}/tools/gen_compilation_database.py" --include_headers ${COMP_DB_TARGETS:+${COMP_DB_TARGETS[@]}}
```

Comment:
im not sure but i think the array expansion wont work here - ill check further

Reply:
arrays arent carried in the env vars so there isnt a way that "${COMP_DB_TARGETS[@]}" would ever be evaluated as one (unless either this script was sourced, or sourced another script that set this)

Reply:
I set this in an env-var with 4 targets and tested it (as above in PR description) and it worked. Are you saying that if I set and exported the value, you would not expect it to work?

Reply:
i created a file with the following:
```
#!/bin/bash

FOO=${COMP_DB_TARGETS:+${COMP_DB_TARGETS[@]}}

echo "$FOO"
```

and then did the following:

```
$ export COMP_DB_TARGETS="xyz"
$ ./runfoo
xyz
$ export COMP_DB_TARGETS=("abc" "xyz")
$ ./runfoo
```

```

Reply:
if `COMP_DB_TARGETS` is an array it wont work

i think you need to do something like

```
read -ra COMP_DB_TARGETS <<< "$COMP_DB_TARGETS"
```

if you want to use an array

Reply:
and then you would also need to set it to a string

Reply:
I have this currently working in our internal CI system with the following:
```bash
COMP_DB_TARGETS="@envoy//source/... @envoy//test/... @envoy//tools/... //filters/..."
export COMP_DB_TARGETS

bash -ex ./envoy/ci/do_ci.sh 'bazel.clang_tidy' '//filters/...'
```

And then with -x on `run_clang_tidy.sh` I see:

```bash
+ /src/envoy/tools/gen_compilation_database.py --include_headers @envoy//source/... @envoy//test/... @envoy//tools/... //filters/...
```

Reply:
its set to a string - my point is `$COMP_DB_TARGETS[@]" is wrong

Reply:
ah, gotcha. I'll update to not do any array expansion.

Reply:
tbh i think you do want array expansion
i added a py script (similar to what is happening in this file)

```
export COMP_DB_TARGETS="abc xyz"
$ ./runfoo
abc xyz
['./runfoo.py', 'abc xyz']
```

if you want the args to be separate at least